src/kernel/clock.c

```diff
@@ -35,3 +35,26 @@ void clock_adjust(timestamp wallclock_now, s64 temp_cal, timestamp sync_complete
     rtc_settimeofday(sec_from_timestamp(wallclock_now));
 }
 KLIB_EXPORT(clock_adjust);
+
+closure_function(0, 1, boolean, timer_id_rtc,
+                timer, t)
+{
+    switch (t->id) {
```

Comment:
Really a nit, but it seems like it would be more straightforward to do the timer adjustment right here, and the timer_adjust below replaced with the pqueue_walk followed by a pqueue_reorder. But maybe not if timer_adjust is likely to be used elsewhere.

Reply:
Yeah, it doesn't need to be this elaborate. Fixed, thanks!